codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/MiddlewareBuilder.java

```diff
@@ -299,10 +299,20 @@ private Collection<? extends ClientConfig> getDefaultClientConfig() {
         ClientConfig logLevel = (new ClientConfig.Builder())
                 .name("log_level")
                 .type("Symbol")
-                .defaultValue(":info")
+                .defaultPrimitiveValue(":info")
                 .documentation("The default log level to use with the Logger.")
                 .build();
 
-        return Arrays.asList(logger, logLevel);
+        ClientConfig plugins = (new ClientConfig.Builder())
```

Comment:
Side note, I think we need a way in application transport to register config that is NOT tied to middleware, kinda like this.

Reply:
It can be registered through the client fragments for request or response as well.  I guess - what are the needs for adding transport specific config thats not connected to request/response middleware?  In general - we shouldn't need to add config that not used by one of those I think.

But if there is a need - it would be easy to add a list of ClientConfig to the ApplicationTransport class.

Reply:
I ran into the issue with logger, which I eventually moved here. Sometimes there's a config that might be used by more than one middleware. Registering it to one doesn't make too much sense and annoying to find IMO